endpoints/openrtb2/amp_auction_test.go

```diff
@@ -122,6 +123,131 @@ func TestAMPPageInfo(t *testing.T) {
 	assert.Equal(t, "test.somepage.co.uk", exchange.lastRequest.Site.Domain)
 }
 
+func TestConsentThroughEndpoint(t *testing.T) {
+	// gdpr consent string that will come inside our http.Request query
+	const consentString = "BOa71ZYOa71ZYAbABBENA8-AAAAbN7_______9______9uz_Gv_r_f__33e8_39v_h_7_-___m_-3zV4-_lvR11yPA1OrfIrwFhiAw"
+
+	// Generate a marshaled openrtb.BidRequest that DOESN'T come with a gdpr consent string
+	fullMarshaledBidRequest, err := getTestBidRequest("", "digitrustId")
+	if err != nil {
+		t.Fatalf("Failed to marshal the complete openrtb.BidRequest object %v", err)
+	}
+
+	stored := map[string]json.RawMessage{
+		"1": json.RawMessage(fullMarshaledBidRequest),
+	}
+
+	theMetrics := pbsmetrics.NewMetrics(metrics.NewRegistry(), openrtb_ext.BidderList())
+	exchange := &mockAmpExchange{}
+
+	endpoint, _ := NewAmpEndpoint(
+		exchange,
+		newParamsValidator(t),
+		&mockAmpStoredReqFetcher{stored},
+		empty_fetcher.EmptyFetcher{},
+		&config.Configuration{MaxRequestSize: maxSize},
+		theMetrics,
+		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
+		map[string]string{},
+		[]byte{},
+		openrtb_ext.BidderMap,
+	)
+	request := httptest.NewRequest("GET", fmt.Sprintf("/openrtb2/auction/amp?tag_id=1&gdpr_consent=%s", consentString), nil)
+	recorder := httptest.NewRecorder()
+	endpoint(recorder, request, nil)
+
+	// Assert our bidRequest was valid
+	if !assert.NotNil(t, exchange.lastRequest, "Endpoint responded with %d: %s", recorder.Code, recorder.Body.String()) {
+		return
+	}
+	// Assert our bidRequest had a valid "User" field
+	if !assert.NotNil(t, exchange.lastRequest.User) {
+		return
+	}
+	// Assert our bidRequest had a valid "User.Ext" field
+	if !assert.NotNil(t, exchange.lastRequest.User.Ext) {
+		return
+	}
+
+	re := regexp.MustCompile(`\"consent\":\".*\"}$`)
+
+	// Assert string `consent` wasn't found in the `fullMarshaledBidRequest` at all
+	assert.Nil(t, re.Find(fullMarshaledBidRequest))
+
+	// Assert string `consent` is found in the User.Ext at all
+	assert.NotNil(t, re.Find([]byte(exchange.lastRequest.User.Ext)))
+
+	// Assert the last request has a valid User object with a consent string equal to that on the URL query
+	var ue openrtb_ext.ExtUser
+	err = json.Unmarshal(exchange.lastRequest.User.Ext, &ue)
+	assert.NoError(t, err)
+
+	// Assert consent string found in `http.Request` was passed correctly to the `User.Ext` object
+	assert.Contains(t, string(request.URL.RawQuery), consentString)
+	assert.Equal(t, consentString, ue.Consent)
+
+	// Assert other user properties found originally in our bid request such as `DigiTrust` were not overwritten
+	assert.Equal(t, "digitrustId", ue.DigiTrust.ID)
```

Comment:
A better approach here would be to do `assert.Equal(t, fullMarshaledBidRequest.User.Ext.DigiTrust.ID, ue.DigiTrust.ID)` rather than using the hardcoded string. Also, would be good to add some error messages for all these asserts.